### PR TITLE
embeded manifest.yaml into install operator cmd

### PR DESCRIPTION
**Pull request by jasmingacic**
# Summary of this proposal

This PR embeds manifest.yaml into install operator cmd. 

## Asana task(s) link.

https://app.asana.com/0/1204316201844829/1205521973956070/f

## Notes for the reviewer

The step in the CI is meant to pull the manifest from core-operator latest release and embed it into  cli command.

@patrick-stephens I'd love to hear your input if this path with gh CLI is good or there is a better way

### CODE REVIEW

File: cmd/operator/install.go [hunk 1]
```diff
@@ -2,6 +2,8 @@ package operator
 
 import (
 	"context"
+	"embed"
+	_ "embed"
```
Comment (patrick-stephens) [inline cmd/operator/install.go:6]: Duplicate?

File: cmd/operator/install.go [hunk 2]
```diff
@@ -148,14 +153,12 @@ func extractDeployment(yml string) (string, error) {
 }
 
 func prepareInstallManifest(coreDockerImage, coreInstanceVersion, namespace string, createNamespace bool) (string, error) {
-	file, err := k8s.GetOperatorManifest(coreInstanceVersion)
+	file, err := f.ReadFile("manifest.yaml")
```
Comment (patrick-stephens) [inline cmd/operator/install.go:156]: Magic string so probably should be a constant

File: cmd/operator/install.go (hunk 1 above)
Comment (jasmingacic) [reply cmd/operator/install.go:6]: The first one is to import "embed" package and the other one is to initialize 
```
//go:embed manifest.yaml
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -191,6 +191,14 @@ jobs:
           gpg_private_key: ${{ secrets.CALYPTIA_GPG_KEY }}
           passphrase: ${{ secrets.CALYPTIA_GPG_KEY_PASSPHRASE }}
 
+      - name: Get the manifest
+        run: |
+          version=$(curl -s https://api.github.com/repos/calyptia/core-operator-releases/releases | jq .[0].tag_name)
+          gh release download -p manifest.yaml -R calyptia/core-operator $version
+          mv manifest.yaml cmd/operator
```
Comment (jasmingacic) [inline .github/workflows/ci.yml:198]: @patrick-stephens is this the best way to get manifest.yaml and move it to cmd/operator. Embed package, for whatever reason by design, only accepts full path and can't use relative path to embed files.

Comment (patrick-stephens) [reply .github/workflows/ci.yml:198]: Depends, we could also checkout the repo or do something like that.

What's the goal though, which version of the manifest do you want? Is it always latest release or a specific release?

Comment (patrick-stephens) [reply .github/workflows/ci.yml:198]: Probably want a `curl -sSfL` just to fail explicitly and handle redirects (shouldn't be any).

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -191,6 +191,13 @@ jobs:
           gpg_private_key: ${{ secrets.CALYPTIA_GPG_KEY }}
           passphrase: ${{ secrets.CALYPTIA_GPG_KEY_PASSPHRASE }}
 
+      - name: Get the manifest
+        run: |
+          version=$(curl -s https://api.github.com/repos/calyptia/core-operator-releases/releases | jq .[0].tag_name | sed -E 's/.*"([^"]+)".*/\1/')
+          curl -sSfL https://github.com/calyptia/core-operator-releases/releases/download/${version}/manifest.yaml -o ./cmd/operator/manifest.yaml
```
Comment (jasmingacic) [inline .github/workflows/ci.yml:197]: @patrick-stephens is this better?

Comment (patrick-stephens) [reply .github/workflows/ci.yml:197]: no idea, does it work? :)

This feels bad though to only add in CI, should there not be a local file for it we can just PR?

Comment (jasmingacic) [reply .github/workflows/ci.yml:197]: Local file is required for local builds. So yes there is one. 
Maybe it is better to do this like we do when we release core-operator when we create PR in charts repo.

Comment (patrick-stephens) [reply .github/workflows/ci.yml:197]: yeah I think so, then we can review it works in the PR too

File: Makefile [hunk 5]
```diff
@@ -2,5 +2,11 @@ VERSION ?= $(shell git describe --tags)
...
-build:
-	go build -ldflags="${LD_FLAGS}" -o calyptia 
\ No newline at end of file
+build: get-manifest
+	go build -ldflags="${LD_FLAGS}" -o calyptia 
+
+get-manifest: 
+	version=$(curl -s https://api.github.com/repos/calyptia/core-operator-releases/releases | jq .[0].tag_name)
+	echo ${version}
+	gh release download -p manifest.yaml -R calyptia/core-operator ${version}
```
[the remark this answers is not in the document]
Comment (jasmingacic) [reply Makefile:11]: yes it does it only needs GITHUB_TOKEN env var to be set.

[the remark this answers is not in the document]
Comment (patrick-stephens) [reply Makefile:11]: We should probably document that in a comment.
Not everyone may have it locally if they want to do this.

[the remark this answers is not in the document]
Comment (jasmingacic) [reply Makefile:11]: i'll ditch that make target

**Review by patrick-stephens — DISMISSED** [review]

**Comment by jasmingacic** [comment]
Needs https://github.com/calyptia/core-operator/pull/275

**Review by patrick-stephens — APPROVED** [review]

**Comment by patrick-stephens** [comment]
@jasmingacic what updates the manifest?

**Comment by jasmingacic** [comment]
> @jasmingacic what updates the manifest?

https://github.com/calyptia/core-operator/blob/main/.github/workflows/release.yaml#L260

**Pull request merged**